akka-actor/src/main/scala/akka/actor/Extension.scala

```diff
@@ -89,6 +94,18 @@ trait ExtensionId[T <: Extension] {
    */
   def get(system: ActorSystem): T = apply(system)
 
+  /**
+   * Returns an instance of the extension identified by this ExtensionId instance.
+   * Java API
+   * For extensions written in Scala that are to be used from Java also,
+   * this method should be overridden to get correct return type.
+   * {{{
+   * override def get(system: ActorSystem): TheExtension = super.get(system)
+   * }}}
+   *
+   */
+  def get(system: ClassicActorSystemProvider): T = apply(system)
```

Comment:
I don't get how this is not an ambiguous overload in at Java, but I see below that you are actually calling it from Java code...

Reply:
For typed ActorSystems only this one matches. For untyped ActorSystems both would match but the old one would be more specific and chosen for that reason.